core/utils.ts

```diff
@@ -293,9 +295,36 @@ export function actionConfigToCompiledGraphTarget(
   if (actionConfigTarget.project) {
     compiledGraphTarget.database = actionConfigTarget.project;
   }
+  if (actionConfigTarget.hasOwnProperty("includeDependentAssertions")) {
+    compiledGraphTarget.includeDependentAssertions = actionConfigTarget.includeDependentAssertions;
+  }
   return dataform.Target.create(compiledGraphTarget);
 }
 
 export function resolveActionsConfigFilename(configFilename: string, configPath: string) {
   return Path.normalize(Path.join(Path.dirName(configPath), configFilename));
 }
+
+export function addDependenciesToActionDependencyTargets(action: actionsWithDependencies, resolvable: Resolvable) {
+  let dependencyTarget = resolvableAsTarget(resolvable);
+  if (!dependencyTarget.hasOwnProperty("includeDependentAssertions")) {
+    // dependency `includeDependentAssertions` takes precedence over the config's `dependOnDependencyAssertions`
+    dependencyTarget.includeDependentAssertions = action.dependOnDependencyAssertions;
+  }
+
+  // check if same dependency already exist in this action but with opposite value for includeDependentAssertions
+  const conflictingIncludeDependentAssertions =
+    action.proto.dependencyTargets.filter(existingDependency =>
```

Comment:
@Ekrekr, is it ok to have a quadratic complexity from the number of action dependencies here? I assume with thousands of actions it may be acceptable, but if in the future we'll scale to tens of thousands potential dependencies it may become quite slow (though the workflows where one action has so many dependencies are weird).

Reply:
Where does the quadratic complexity come from? I don't really follow

Reply:
replaced this loop with Map to make it O(n).